daemon/healthz/kube_proxy_healthz.go

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package healthz
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net"
+	"net/http"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+	"github.com/spf13/pflag"
+	"golang.org/x/sys/unix"
+
+	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/service"
+	"github.com/cilium/cilium/pkg/status"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+// ServiceInterface to help with testing.
+type ServiceInterface interface {
+	GetLastUpdatedTs() time.Time
```

Comment:
Note to myself that I can link to the LB tracking issue: this is currently going via `(*serviceManagerAdapter).GetLastUpdatedTs` which just returns `time.Now()`.  Once we can remove the adapters we can update this to just look at the timestamp of the last object in the frontends table which either requires adding reverse iterators to StateDB, or can also do `Revision()` + lookup from the revision index.